evm/loomevm.go

```diff
@@ -121,20 +127,22 @@ type LoomVm struct {
 	receiptHandler loomchain.WriteReceiptHandler
 	createABM      AccountBalanceManagerFactoryFunc
 	debug          bool
+	tracer         ethvm.Tracer
 }
 
 func NewLoomVm(
 	loomState loomchain.State,
-	eventHandler loomchain.EventHandler,
 	receiptHandler loomchain.WriteReceiptHandler,
 	createABM AccountBalanceManagerFactoryFunc,
 	debug bool,
+	tracer ethvm.Tracer,
```

Comment:
Since the tracer is optional add a new function to set it instead of adding it to the constructor:
```golang
func (lvm LoomVm) WithTracer(tracer ethvm.Tracer) LoomVm {
  lvm.tracer = tracer
  return lvm
}
```
And it doesn't look like the `debug` parameter is ever set to anything other than `false`, so I think you can remove it from the construction function as well.

Reply:
The evm debugging flag could be considered redundant with the new trace functionality. And as it is not actually used, do you want me to remove the evmDebugging functionality - which is just set a trace log - can be done with tracer - and write out a warning.

Reply:
NewLoomVm returns a `vm.VM` not a `LoomVm` object so we need to change something to use `loomVm.WithTracer`. Probably best to change the return type of `NewLoomVm` to `LoomVm` so SetTracer can be run on it. Could also add `WithTracer` to VM interface.

Reply:
> The evm debugging flag could be considered redundant with the new trace functionality and could be removed. It sets a trace log - can be done with tracer - and writes out a warning.
> Evm Debug can be set from loom.yaml at https://github.com/loomnetwork/loomchain/blob/master/cmd/loom/loom.go#L837 . I can replace the config `EVMDebugEnabled` with an `EVMTraceEnable` option which can add a trace to the `main app.TxHandlerFactory` and dumps the result to the log file?

Yeah, the EVM debug flag doesn't seem to actually do anything unless the tracer is set, so lets replace it with an `EVMTracerEnabled` setting.